core/webviz/_page_element.py

```diff
@@ -51,5 +59,11 @@ def __str__(self):
                       type='text/css'>
                 </link>\n""".format(css)
         for js in js_deps:
-            html += "<script src='{0}'></script>\n".format(js)
+            if isinstance(js, StringIO):
+                html += "<script>{0}</script>\n".format(js.getvalue())
```

Comment:
It is probably not a good idea to check whether this is `StringIO`. Generally, js can be any stream or a filename.